woodwork/type_sys/type_system.py

```diff
@@ -315,11 +315,22 @@ def get_inference_matches(types_to_check, series, type_matches=[]):
                 get_inference_matches(check_next, series, type_matches)
             return type_matches
 
-        type_matches = get_inference_matches(self.root_types, series)
+        # Don't include NaturalLanguage as we only want to check that if
+        # no other matches are found
+        types_to_check = [
+            ltype for ltype in self.root_types if ltype != NaturalLanguage
```

Comment:
Can we change the order of `self.root_types` to have NaturalLanguage be the last value?